tests/test_connection_manager.c

```diff
@@ -161,13 +164,12 @@ void s_release_connections(size_t count, bool close_first) {
     }
 
     if (release_count == 0) {
-        return;
+        goto release;
```

Comment:
Does this silence actual errors? Or just make the test resilient to expected periodic failures?

Reply:
It should be just resilience.  The connection requests are for real targets so they may fail occasionally, which would result in nothing to release.